src/riot-rs-threads/src/channel.rs

```diff
@@ -92,33 +92,29 @@ impl<T: Copy + Send> Channel<T> {
     }
 
     pub fn recv(&self) -> T {
-        let mut res = MaybeUninit::uninit();
+        let uc = UnsafeCell::<MaybeUninit<T>>::new(MaybeUninit::uninit());
```

Comment:
Why is the `UnsafeCell` needed?

Reply:
I'm not sure it is. My thinking was that I'm passing around a reference to `&res`, and I don't want the compiler to optimize around the assumption that I'm not actually modifying it in that function.

Reply:
Based on the [docs](https://doc.rust-lang.org/core/mem/union.MaybeUninit.html) I don't the compiler does any optimization for `MaybeUninit`:
> MaybeUninit<T> serves to enable unsafe code to deal with uninitialized data. It is a signal to the compiler indicating that the data here might not be initialized
>
> The compiler then knows to not make any incorrect assumptions or optimizations on this code.

Reply:
I dropped the "UnsafeCell" wrapping.